src/features/smartSearch/components/StyledMsg.tsx

```diff
@@ -0,0 +1,48 @@
+import { useIntl } from 'react-intl';
+
+import UnderlinedText from './UnderlinedText';
+import {
+  InterpolatedMessage,
+  PlainMessage,
+  ValueRecord,
+} from 'core/i18n/messages';
+
+type StyledPlainMsgProps = {
+  id: PlainMessage;
+  values?: void;
+};
+
+type StyledInterpolatedMsgProps<Values extends ValueRecord> = {
+  id: InterpolatedMessage<Values>;
+  values: Values;
+};
+
+type StyledMsgProps<Values extends ValueRecord> = {
+  id: PlainMessage | InterpolatedMessage<Values>;
+  values?: Values;
+};
+
+function StyledMsg({ id, values }: StyledPlainMsgProps): JSX.Element;
```

Comment:
Would it maybe make sense to rename this to `UnderlinedMsg` now to be consistent with the `UnderlinedText` component?

Reply:
Sounds like a good idea!